### PR TITLE
feat: package vllm runtime into image

### PR DESCRIPTION
**Pull request by zhuangqh**
**Reason for Change**:

- pack vllm runtime and chat_template files
- align the requirements of the two runtimes
- support to load chat_template for hf runtime

**Requirements**

- [ ] added unit tests and e2e tests (if applicable).

**Issue Fixed**:

**Notes for Reviewers**:

### CODE REVIEW

File: presets/requirement/README.md [hunk 1]
```diff
@@ -0,0 +1,3 @@
+# Requirement
+
+Pip dependencies for both Kaito inference and tuning.
```
Comment (bangqipropel) [inline presets/requirement/README.md:3]: For here, can we add some more instructions to clarify this is about vllm? From the path to the content I cannot see the scope of this Requirement

File: Makefile [hunk 2]
```diff
@@ -114,8 +114,7 @@ tuning-metrics-server-test: virtualenv
 ## --------------------------------------
 
 inference-api-e2e: virtualenv
-	./hack/run-pytest-in-venv.sh presets/inference/vllm presets/inference/vllm/requirements.txt
-	./hack/run-pytest-in-venv.sh presets/inference/text-generation presets/inference/text-generation/requirements.txt
+	./hack/run-pytest-in-venv.sh presets/inference presets/requirement/requirements-test.txt
```
Comment (ishaansehgal99) [inline Makefile:117]: not using virtualenv anymore

File: presets/requirement/README.md (hunk 1 above)
Comment (zhuangqh) [reply presets/requirement/README.md:3]: it works for both vllm and transformers.

**Review by Fei-Guo — APPROVED** [review]

File: presets/inference/text-generation/inference_api.py [hunk 3]
```diff
@@ -492,7 +514,11 @@ def get_metrics():
         logger.error(f"Error fetching metrics: {e}")
         raise HTTPException(status_code=500, detail=str(e))
 
+def shutdown_handler(sig, frame):
+    sys.exit(0)
+
 if __name__ == "__main__":
+    signal.signal(signal.SIGINT, shutdown_handler)
```
Comment (ishaansehgal99) [inline presets/inference/text-generation/inference_api.py:521]: why shutdown sig?

File: docker/presets/models/tfs/Dockerfile [hunk 4]
```diff
@@ -5,31 +5,29 @@ ARG MODEL_TYPE
...
-# Copy the metrics server
-COPY kaito/presets/tuning/${MODEL_TYPE}/metrics/metrics_server.py /workspace/tfs/metrics_server.py
-
-# Copy the entire model weights to the weights directory
-COPY ${WEIGHTS_PATH} /workspace/tfs/weights
+# Model weights
+COPY ${WEIGHTS_PATH} /workspace/weights
+RUN echo $VERSION > /workspace/version.txt && \
+    ln -s /workspace/weights /workspace/tfs/weights && \
+    ln -s /workspace/weights /workspace/vllm/weights
```
Comment (ishaansehgal99) [inline docker/presets/models/tfs/Dockerfile:33]: nice

File: presets/inference/text-generation/inference_api.py [hunk 5]
```diff
@@ -98,7 +117,10 @@ def __post_init__(self): # validate parameters
 combination_type = model_args.pop('combination_type')
 
 app = FastAPI()
+resovled_chat_template = load_chat_template(model_args.pop('chat_template'))
```
Comment (ishaansehgal99) [inline presets/inference/text-generation/inference_api.py:120]: so just to make sure, now the filename to the chat template is passed in as a default param, we will need to update this on the controller side in future PR?

File: docker/presets/models/tfs/Dockerfile [hunk 6]
```diff
@@ -5,31 +5,29 @@ ARG MODEL_TYPE
...
-RUN pip install --no-cache-dir -r tuning-requirements.txt
+# 2. vLLM
+COPY kaito/presets/inference/vllm/inference_api.py /workspace/vllm/inference_api.py
 
-COPY kaito/presets/tuning/${MODEL_TYPE}/cli.py /workspace/tfs/cli.py
-COPY kaito/presets/tuning/${MODEL_TYPE}/fine_tuning.py /workspace/tfs/fine_tuning.py
-COPY kaito/presets/tuning/${MODEL_TYPE}/parser.py /workspace/tfs/parser.py
-COPY kaito/presets/tuning/${MODEL_TYPE}/dataset.py /workspace/tfs/dataset.py
+# Chat template
+ADD kaito/presets/inference/chat_templates /workspace/chat_templates
```
Comment (ishaansehgal99) [inline docker/presets/models/tfs/Dockerfile:27]: coooool

**Review by ishaansehgal99 — APPROVED** [review]

File: presets/inference/text-generation/inference_api.py (hunk 3 above)
Comment (ishaansehgal99) [reply presets/inference/text-generation/inference_api.py:521]: nvm I see this just registers the signal, looks good

**Pull request merged**